### PR TITLE
Issue #126

### PR DESCRIPTION
**Pull request by jinglemansweep**
Added bytearray('\x00') to the list of expected "noop" returns from the socket.

### CODE REVIEW

File: adafruit_minimqtt/adafruit_minimqtt.py [hunk 1]
```diff
@@ -896,7 +896,7 @@ def _wait_for_msg(self, timeout=0.1):
 
         # Block while we parse the rest of the response
         self._sock.settimeout(timeout)
-        if res in [None, b""]:
+        if res in [None, b"", bytearray(b'\x00')]:
```
Comment (dhalbert) [inline adafruit_minimqtt/adafruit_minimqtt.py:899]: Does it work if this is just `b"\x00"`? Not sure you need to create a bytearray.

**Review by dhalbert — CHANGES_REQUESTED** [review]
THanks! One question.

**Comment by dhalbert** [comment]
Use `pre-commit` or run the code through `black` to get rid of the failure: https://learn.adafruit.com/improve-your-code-with-pylint

File: adafruit_minimqtt/adafruit_minimqtt.py (hunk 1 above)
Comment (jinglemansweep) [reply adafruit_minimqtt/adafruit_minimqtt.py:899]: I've just tried:

    if res in [None, b"", b"\x00"]:

And it appears to work as expected. Decent matrix framerates, no `MemoryError`'s etc.

**Review by dhalbert — APPROVED** [review]
I'm wondering whether the fact that you have to check for one byte of zeros is a bug in the underlying network code. But I think this is a good workaround to add to the library for now.

@brentru Do you have any comment about this from an MQTT protocol point of view?

**Comment by brentru** [comment]
@jinglemansweep Hi - thanks for fixing this issue. @dhalbert - I do not think this is a network or a MQTT spec issue, `b"\x00` is rx'd after the call to `res = self._sock_exact_recv(1)`. 

Considering some in #126 are having issues with non-native wifi,  it's possible (don't know, guessing right now) that the issue is within this library's local implementation of `_sock_exact_recv()` vs the native impl. of `_sock_exact_recv()` (cc @calcut ☝️ )

```
        # CPython socket module contains a timeout attribute
        if hasattr(self._socket_pool, "timeout"):
            try:
                res = self._sock_exact_recv(1)
            except self._socket_pool.timeout:
                return None
        else:  # socketpool, esp32spi
            try:
                res = self._sock_exact_recv(1)
```

**Review by brentru — APPROVED** [review]
For now, though, I think this is an OK temporary change and will cause clients to poll faster until we resolve why the socket is rcv'ing a bytearray on some clients.

**Pull request merged**